backend/services/implementations/testSessionService.ts

```diff
@@ -187,6 +204,78 @@ class TestSessionService implements ITestSessionService {
 
     return testSessionDtos;
   }
+
+  /*
+   * computeTestGrades computes the breakdown and score of a given result and returns the ResultDTO
+   */
+  async computeTestGrades(
+    result: NewResultDTO,
+    testId: string,
+  ): Promise<ResultDTO> {
+    let resultDto: ResultDTO;
+    let questionsCorrect = 0; // the number of questions the student gets correct
+    const studentAnswers: number[] = result.answers; // the array of the student's raw answers
+    let updatedScore = 0.0; // the final score that will be returned
+    const updatedBreakdown: boolean[] = []; // the final breakdown that will be returned
+
+    try {
+      const test: TestResponseDTO = await this.testService.getTestById(testId);
+
+      // check if the number of test questions matches the number of student answers
+      if (test.questions.length !== studentAnswers.length) {
```

Comment:
Not sure if this check is necessary, as discussed [here](https://github.com/uwblueprint/jump-math/pull/40#discussion_r932817119). Let me know if I'm missing something!